src/library_syscall.js

```diff
@@ -300,6 +300,15 @@ var SyscallsLibrary = {
       if (info.allocated) {
         _free(info.malloc);
       }
+    } else {
+      // len < info.len, which means a partial munmap (i.e., not of the entire
+      // allocated region). If we did an actual allocation, this would leak, as
+      // all we have is malloc/free and not a real mmap.
+#if ASSERTIONS
```

Comment:
How about doing this above and then unidenting the rest of the function

```
if (len !== info.len) {
#if ASSERTIONS
  err('munmap of part of an mmap-ed region is not supported, and will leak');
#endif
  # TODO: we probably shouldn't be returning success here since we didn't actually free unmap anything.
  return 0;
}
```

I prefer that because it keeps the check and the error in the same place and reduces indentation, as well as highlighting that we are returning zero despite the error.

Reply:
Sure, done.